### ISSUE TITLE
Project-specific latexmkrc is not executed

### ISSUE THREAD

**Issue by kiryph** [lervag/vimtex#1388]
**Describe the issue**
Project-specific latexmkrc is not executed. To verify this, I considered a simple latexmkrc file with a print statement. A minimal test already exists in  [`vimtex/test/features/latexmk-internal`](https://github.com/lervag/vimtex/tree/master/test/features/latexmk-internal).

**Steps to reproduce**

Preliminary step:
I have renamed `latexmk.rc` in `vimtex/test/features/latexmk-internal` to `latexmkrc` according to http://mirrors.ctan.org/support/latexmk/latexmk.pdf#page=19

> 3) The RC file in the current working directory. This file can be named either "latexmkrc" or  ".latexmkrc", and the first of these to be found is used, if any.

Without renaming the file to `latexmkrc` latexmk does not consider the file `latexmk.rc`.

Now I have in the folder `vimtex/test/features/latexmk-internal`:
```
❯ cat latexmkrc
$pdflatex = 'internal internalexample %R %O %S';

sub internalexample {
    my @args = @_;
    print "This line should be printed\n";
    return system("pdflatex @args");
}

❯ cat test-internal.tex
\documentclass[11pt,a4paper]{article}

\begin{document}

This is just a simple example. Compiling it with the following \texttt{.latexmkrc} should print ``This line should be printed'' before starting \texttt{pdflatex}.

\begin{verbatim}
$pdflatex = 'internal internalexample %R %O %S';

sub internalexample {
    my @args = @_;
    print "This line should be printed\n";
    return system("pdflatex @args");
}
\end{verbatim}

\end{document}
```

When I run it on the command line with `$ latexmk test-internal.tex`, I see the line `This line should be printed`:
```
Latexmk: This is Latexmk, John Collins, 17 March 2019, version: 4.63b.
Latexmk: applying rule 'pdflatex'...
Rule 'pdflatex': The following rules & subrules became out-of-date:
      'pdflatex'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'internal internalexample "test-internal"  -recorder  "test-internal.tex"'
------------
Latexmk: calling internalexample( "test-internal"  -recorder  "test-internal.tex" )
This line should be printed
This is pdfTeX, Version 3.14159265-2.6-1.40.20 (TeX Live 2019) (preloaded format=pdflatex)
 restricted \write18 enabled.
```

When I compile the file `test-internal.tex` in vim (`$ vim test-internal.tex`) with vimtex `<leader>ll`, the log file does not have this line (tested with the correct filename`latexmkrc`).
I get
```
Latexmk: This is Latexmk, John Collins, 17 March 2019, version: 4.63b.
Viewing pdf
Rule 'pdflatex': The following rules & subrules became out-of-date:
      'pdflatex'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'internal internalexample "test-internal"  -file-line-error -synctex=1 -interaction=nonstopmode -recorder  "test-internal.tex"'
------------
Latexmk: calling internalexample( "test-internal"  -file-line-error -synctex=1 -interaction=nonstopmode -recorder  "test-internal.tex" )
Latexmk: fls file doesn't appear to have been made.
Latexmk: Log file says output to 'test-internal.pdf'
Latexmk: I have not found a previewer that is already running. 
   So I will start it for 'test-internal.pdf'
------------
------------
Running 'start open  "test-internal.pdf"'
------------
------------
Running 'echo vimtex_compiler_callback_success'
------------
cle.cls
Document Class: article 2018/09/03 v1.4i Standard LaTeX document class
(/usr/local/texlive/2019/texmf-dist/tex/latex/base/size11.clo)) (./test-internal.aux)
Overfull \hbox (3.8917pt too wide) in paragraph at lines 5--6
[]\OT1/cmr/m/n/10.95 This is just a sim-ple ex-am-ple. Com-pil-ing it with the fol-low-ing \OT1/cmtt/m/n/10.95 .latexmkrc
[1{/Users/kiryph/Library/texlive/2019/texmf-var/fonts/map/pdftex/updmap/pdftex.map}] (./test-internal.aux) )
(see the transcript file for additional information)</usr/local/texlive/2019/texmf-dist/fonts/type1/public/amsfonts/cm/cmr10.pfb></usr/local/texlive/2019/texmf-dist/fonts/type1/public/amsfonts/cm/cmtt10.pfb>
Output written on test-internal.pdf (1 page, 30618 bytes).
Transcript written on test-internal.log.
=== TeX engine is 'pdfTeX'
Latexmk: All targets () are up-to-date
For rule 'view', running '&if_source(  )' ...
vimtex_compiler_callback_success
```

My system info
```
System info
  OS: Mac OS X 10.14.5 (18F132)
  Vim version: VIM 8.1 (1-950)
  Has clientserver: true
  Servername: VIM1

vimtex project: test-internal
  base: test-internal.tex
  root: /Users/kiryph/.vim-configs/devel-vimtex/pack/vimtex/start/vimtex/test/features/latexmk-internal
  tex: /Users/kiryph/.vim-configs/devel-vimtex/pack/vimtex/start/vimtex/test/features/latexmk-internal/test-internal.tex
  out: /Users/kiryph/.vim-configs/devel-vimtex/pack/vimtex/start/vimtex/test/features/latexmk-internal/test-internal.pdf
  log: /Users/kiryph/.vim-configs/devel-vimtex/pack/vimtex/start/vimtex/test/features/latexmk-internal/test-internal.log
  aux: /Users/kiryph/.vim-configs/devel-vimtex/pack/vimtex/start/vimtex/test/features/latexmk-internal/test-internal.aux
  fls: 
  compiler: latexmk
    backend: jobs
    output: /var/folders/d_/5kjxh9v92k752bvzf559fd3c0000gn/T/vUlbFIc/0
    configuration: 
      continuous: 1
      callback: 1
      latexmk options:
        -verbose
        -file-line-error
        -synctex=1
        -interaction=nonstopmode
      latexmk engine: -pdf
  viewer: General
  qf: LaTeX logfile
    config: 
      fix_paths: 1
      packages: 
        default: 1
      default: 1
  document class: article
```

It feels like this should have been solved when I see related issues.

Do I overlook something?

What is the build directory in general?

**Comment by lervag** [comment]
Sorry for taking so long! I've tested this now. I found that with neovim, this problem does not exist. It is only with Vim. In a way, it is clear that the latexmkrc file is sourced, so that is not the problem. Instead, I believe it might be something with the code that reads the output with `\lo`. I assume this is what you mean when you say "log file".

I'll look into it, but I'm curious: Do you have any _real_ problems with this, or is it only a curiosity?

**Comment by lervag** [comment]
Ok, I think it should be fixed now. Thanks for reporting!

**Comment by kiryph** [comment]
> Do you have any real problems with this, or is it only a curiosity?

I was debugging the file [`pythontex-latexmkrc`](https://mirror.informatik.hs-fulda.de/tex-archive/support/latexmk/example_rcfiles/pythontex-latexmkrc) provided by latexmk with print lines which did not show up.

Right now I cannot recreate my actual problem with compiling a document using pythontex and this latexmkrc file. But apparently seeing the output of debug print statements can be helpful.

By trying to find out what my problem with pythontex and its compilation was, I noticed that we could list errors in the quickfix list emitted by python when executed for the python code parts. Following simple file using the above mentioned latexmkrc file

```tex
\documentclass{article}
\usepackage{pythontex}

\begin{document}

\begin{pycode}
    print(r'\begin{center}')
    print(r'\textit{A message from Python!}')
    print(r'\end{center}')
\end{pycode}

\end{document}
```

will print following to the command line. See at the end for the `IndentationError` which is not listed in the quickfix list when run via vimtex in vim
```
❯ latexmk
Latexmk: This is Latexmk, John Collins, 17 March 2019, version: 4.63b.
Latexmk: applying rule 'pdflatex'...
Rule 'pdflatex': The following rules & subrules became out-of-date:
      'pdflatex'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'pdflatex -shell-escape -recorder -synctex=1  -recorder  "demo-pythontex.tex"'
------------
This is pdfTeX, Version 3.14159265-2.6-1.40.20 (TeX Live 2019) (preloaded format=pdflatex)
 \write18 enabled.
entering extended mode
(./demo-pythontex.tex
LaTeX2e <2018-12-01>
(/usr/local/texlive/2019/texmf-dist/tex/latex/base/article.cls
Document Class: article 2018/09/03 v1.4i Standard LaTeX document class
(/usr/local/texlive/2019/texmf-dist/tex/latex/base/size10.clo))
(/usr/local/texlive/2019/texmf-dist/tex/latex/pythontex/pythontex.sty
(/usr/local/texlive/2019/texmf-dist/tex/latex/fvextra/fvextra.sty
(/usr/local/texlive/2019/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/local/texlive/2019/texmf-dist/tex/latex/etoolbox/etoolbox.sty)
(/usr/local/texlive/2019/texmf-dist/tex/latex/fancyvrb/fancyvrb.sty
Style option: `fancyvrb' v3.2a <2019/01/15> (tvz)
(/usr/local/texlive/2019/texmf-dist/tex/latex/graphics/keyval.sty))
(/usr/local/texlive/2019/texmf-dist/tex/latex/upquote/upquote.sty)
(/usr/local/texlive/2019/texmf-dist/tex/latex/lineno/lineno.sty))
(/usr/local/texlive/2019/texmf-dist/tex/generic/xstring/xstring.sty
(/usr/local/texlive/2019/texmf-dist/tex/generic/xstring/xstring.tex))
(/usr/local/texlive/2019/texmf-dist/tex/latex/pgfopts/pgfopts.sty
(/usr/local/texlive/2019/texmf-dist/tex/latex/pgf/utilities/pgfkeys.sty
(/usr/local/texlive/2019/texmf-dist/tex/generic/pgf/utilities/pgfkeys.code.tex
(/usr/local/texlive/2019/texmf-dist/tex/generic/pgf/utilities/pgfkeysfiltered.c
ode.tex)))) (/usr/local/texlive/2019/texmf-dist/tex/latex/caption/newfloat.sty)
 (/usr/local/texlive/2019/texmf-dist/tex/latex/currfile/currfile.sty
(/usr/local/texlive/2019/texmf-dist/tex/latex/oberdiek/kvoptions.sty
(/usr/local/texlive/2019/texmf-dist/tex/generic/oberdiek/ltxcmds.sty)
(/usr/local/texlive/2019/texmf-dist/tex/generic/oberdiek/kvsetkeys.sty
(/usr/local/texlive/2019/texmf-dist/tex/generic/oberdiek/infwarerr.sty)
(/usr/local/texlive/2019/texmf-dist/tex/generic/oberdiek/etexcmds.sty
(/usr/local/texlive/2019/texmf-dist/tex/generic/oberdiek/ifluatex.sty))))
(/usr/local/texlive/2019/texmf-dist/tex/latex/filehook/filehook.sty)))
(/usr/local/texlive/2019/texmf-dist/tex/latex/xcolor/xcolor.sty
(/usr/local/texlive/2019/texmf-dist/tex/latex/graphics-cfg/color.cfg)
(/usr/local/texlive/2019/texmf-dist/tex/latex/graphics-def/pdftex.def))
(./demo-pythontex.aux)
No file pythontex-files-demo-pythontex/demo-pythontex.pytxmcr.
Run PythonTeX to create it.
(/usr/local/texlive/2019/texmf-dist/tex/context/base/mkii/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (./demo-pythontex.aux) )
No pages of output.
Transcript written on demo-pythontex.log.
=== TeX engine is 'pdfTeX'
Latexmk: Missing input file: 'pythontex-files-demo-pythontex/demo-pythontex.pytxmcr' from line
  'No file pythontex-files-demo-pythontex/demo-pythontex.pytxmcr.'
Latexmk: Log file says no output from latex
Latexmk: For rule 'pdflatex', no output was made
Latexmk: applying rule 'pythontex'...
Rule 'pythontex': File changes, etc:
   Changed files, or newly in use since previous run(s):
      'demo-pythontex.pytxcode'
------------
Run number 1 of rule 'pythontex'
------------
For rule 'pythontex', running '&mypythontex(  )' ...
This line should be printed
------------
Running 'pythontex  "demo-pythontex.pytxcode"'
------------
This is PythonTeX 0.16

----  Messages for py:default:default  ----
* PythonTeX stderr - error on line 7:
    File "<outputdir>/py_default_default.py", line 53
      print(r'\begin{center}')
      ^
  IndentationError: unexpected indent

--------------------------------------------------
PythonTeX:  demo-pythontex - 1 error(s), 0 warning(s)

Latexmk: Errors, so I did not complete making targets
Latexmk: Use the -f option to force complete processing,
 unless error was exceeding maximum runs, or warnings treated as errors.
```

I do not know what your plans for vimtex are. I could understand that you want to limit the scope of vimtex with respect to support for LaTeX packages.

**Comment by lervag** [comment]
First, do you agree that the present issue is solved?

If so, then I suggest we close this issue and open a new one for possibly handling PythonTeX error messages. If I understand correctly, it is essentially python errors, but perhaps it would be possible to only parse the "PythonTeX stderr - error on line 7" part? In any case, let's continue that discussion elsewhere.

**Comment by kiryph** [comment]
> First, do you agree that the present issue is solved?

Yes, the issue is resovled. Thank you!

If I read the corresponding commit correctly, `print` in latexmkrc writes to stderr. I find this quite surprising. This is the reason why I was assuming that the project latexmkrc is not executed at all.

Did I get this right?

I open a new issue for parsing python errors when using pythontex. I think it would be nice to have this because otherwise you do not get a feedback when compiling within vimtex a pythontex document with an error in the python part (no pdf is generated).

**Comment by lervag** [comment]
> Did I get this right?

Yes, I think you do. The output by latexmk goes to stderr, while the output from the `pdflatex` (or equivalent command) goes to stdout, it seems.
